ako-gateway-api/nodes/gateway_model.go

```diff
@@ -217,7 +240,12 @@ func AddTLSNode(key string, object *AviObjectGraph, gateway *gatewayv1.Gateway,
 		if listener.TLS != nil {
 			for _, certRef := range listener.TLS.CertificateRefs {
 				name := string(certRef.Name)
-				encodedCertName := lib.GetTLSKeyCertNodeName("", string(*listener.Hostname), name)
+				// Hostname can be empty so initialize it to section name of listener
+				hostname := string(listener.Name)
```

Comment:
can this assumption cause problems with multiple gateways with similar listener names? 

should we change this to gateway+listener or namespace+gateway+listener?

Reply:
This naming convention has been borrowed from AKO ingress container. We need to re-evaluate it as naming convention depends upon, hostname-infrasettingname-certRefName. In Gateway implementation, hostname can be spread across multiple gateway so there could be clash of names. 
This can be taken out as separate PR since it will require bunch of another validations.

Reply:
ok